agenda/presentation/src/main/java/com/nibalk/tasky/agenda/presentation/components/AgendaDetailHeader.kt

```diff
@@ -0,0 +1,125 @@
+package com.nibalk.tasky.agenda.presentation.components
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.nibalk.tasky.agenda.presentation.R
+import com.nibalk.tasky.core.presentation.components.TaskyHeader
+import com.nibalk.tasky.core.presentation.themes.TaskyTheme
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+@Composable
+fun AgendaDetailHeader(
+    modifier: Modifier = Modifier,
+    datePattern: String = "dd MMMM yyyy",
+    isEditable: Boolean,
+    headerDate: LocalDate,
+    headerTitle: String,
+    onCloseDetail: () -> Unit,
+    onSaveDetail: () -> Unit,
+    onIsEditableChanged: (Boolean) -> Unit,
+) {
+    val isEditingMode = remember {
+        mutableStateOf(isEditable)
+    }
```

Comment:
I'd keep such states in the ViewModel's state. Right now, this state wouldn't be restored after process death. You could solve that by using `rememberSaveable`, but there's a reason to have all the states in the ViewModel after all (to keep responsibilities separate) 😄 

Why don't you use the already passed `isEditable` boolean as state for this? Why work with two states instead?